### DISCUSSION TITLE
what's different between use and not use await base.OnAfterRenderAsync

### DISCUSSION THREAD

**Discussion by MahdiElahi** [dotnet/runtime#104602] [Q&A]
what's different between use and not use await base.OnAfterRenderAsync ?
what's different between these codes ?

```
@functions {
    protected async override Task OnAfterRenderAsync(bool firstRender)
    {

        if (!firstRender)
        {
            IsDataLoaded = true;
            StateHasChanged();
        }
        await base.OnAfterRenderAsync(firstRender);

    }
}
```
and

```
@functions {
    protected async override Task OnAfterRenderAsync(bool firstRender)
    {

        if (!firstRender)
        {
            IsDataLoaded = true;
            StateHasChanged();
        }


    }
```

**Comment by stephentoub** [comment]
This isn't really about await or not; it's about whether you're calling to the base method or not. And whether that's impactful would depend on what that base method does. What type are you derived from here? In general, in such an override you should be calling to the base unless you know for a fact it is and will always be a nop.

**Comment by ghost** [comment]
Tagging subscribers to this area: @dotnet/area-meta
See info in [area-owners.md](https://github.com/dotnet/runtime/blob/main/docs/area-owners.md) if you want to be subscribed.
<details>
<summary>Issue Details</summary>
<hr />

what's different between use and not use await base.OnAfterRenderAsync ?
what's different between these codes ?

```
@functions {
    protected async override Task OnAfterRenderAsync(bool firstRender)
    {

        if (!firstRender)
        {
            IsDataLoaded = true;
            StateHasChanged();
        }
        await base.OnAfterRenderAsync(firstRender);

    }
}
```
and

```
@functions {
    protected async override Task OnAfterRenderAsync(bool firstRender)
    {

        if (!firstRender)
        {
            IsDataLoaded = true;
            StateHasChanged();
        }


    }
```

<table>
  <tr>
    <th align="left">Author:</th>
    <td>MahdiElahi</td>
  </tr>
  <tr>
    <th align="left">Assignees:</th>
    <td>-</td>
  </tr>
  <tr>
    <th align="left">Labels:</th>
    <td>

`question`, `area-Meta`, `untriaged`, `needs-area-label`

</td>
  </tr>
  <tr>
    <th align="left">Milestone:</th>
    <td>-</td>
  </tr>
</table>
</details>
